### PR TITLE
📦 [Cache] Fix remote caching of targets with module map

### PR DESCRIPTION
**Pull request by danyf90**
### Short description 📝

When a target with a defined modulemap in the `MODULEMAP_FILE` setting is cached, the modulemap becomes part of the cached xcframework.
As the modulemap contains absolute paths, this breaks caching when the same artifact is used on different machines, and hence at different paths.

An alternative to specifying the `MODULEMAP_FILE` setting on the target requiring the modulemap, is to pass the modulemap file to the targets who depends on it, using the `-fmodule-map-file` compiler flag.

It seems to also fix https://github.com/tuist/tuist/issues/3459

### Checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase.
- [ ] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [ ] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [ ] In case the PR introduces changes that affect users, the documentation has been updated.

### CODE REVIEW

File: Sources/TuistGenerator/Mappers/ModuleMapMapper.swift [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                var mappedTarget = mappedProject.targets[t]
+                let targetID = TargetID(projectPath: mappedProject.path, targetName: mappedTarget.name)
+                var mappedSettingsDictionary = mappedTarget.settings?.base ?? [:]
+                guard mappedSettingsDictionary[Self.modulemapFileSetting] != nil || targetToModuleMaps[targetID] != nil else { continue }
+
+                if mappedSettingsDictionary[Self.modulemapFileSetting] != nil {
+                    mappedSettingsDictionary[Self.modulemapFileSetting] = nil
+                }
+
+                if let dependenciesModuleMaps = targetToModuleMaps[targetID] {
```
Comment (fortmarek) [inline Sources/TuistGenerator/Mappers/ModuleMapMapper.swift:36]: can e.g. the contents of this `if let` be moved a to a `private function`? I feel like this `map` function is kind of hard to digest. At the very least, I'd add some small comments where you would see appropriate

File: Sources/TuistGenerator/Mappers/ModuleMapMapper.swift [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+                mappedTarget.settings = (mappedTarget.settings ?? .default).with(base: mappedSettingsDictionary)
+                mappedProject.targets[t] = mappedTarget
+            }
+            mappedWorkspace.projects[p] = mappedProject
+        }
+        return (mappedWorkspace, [])
+    }
+
+    fileprivate static func dependenciesModuleMaps(
```
Comment (fortmarek) [inline Sources/TuistGenerator/Mappers/ModuleMapMapper.swift:62]:
```suggestion
    private static func dependenciesModuleMaps(
```

Comment (fortmarek) [reply Sources/TuistGenerator/Mappers/ModuleMapMapper.swift:62]: can you add comment to this function as a high-level overview about what it does and why?

**Review by fortmarek — APPROVED** [review]
This code is 🤯 but after careful reading, it makes sense. I'm glad this is contained in a single mapper which makes it much easier to reason about!

**Review by luispadron — APPROVED** [review]
The code in here is above my understanding, but the tests help! Thanks for this

**Comment by fortmarek** [comment]
I fixed the `CHANGELOG.md` conflict, let's merge this once the pipeline is green.

**Pull request merged**